### DISCUSSION TITLE
Release 1.0.0

### DISCUSSION THREAD

**Discussion by TheR1D** [TheR1D/shell_gpt#404] [Announcements]
ShellGPT `v1.0.0` release includes multiple significant changes:
* Default model has been changed to `gpt-4-1106-preview` (a.k.a. GPT-4 Turbo).
* ShellGPT roles (prompts) optimised for OpenAI GPT-4 models.
* Using system roles when calling OpenAI API with [messages](https://platform.openai.com/docs/api-reference/chat/create#chat-create-messages).
* Rendering markdown for default and describe shell command outputs.
* New config variable `CODE_THEME` which sets theme for markdown (default is `dracula`).
* Multiline input in REPL mode possible with `"""` triple quotes.
* New `--version` options that prints installed ShellGPT version.
* Fixed issue with home direcotry in Dockerfile which leading to container crash.
* Code optimisations, minor bug fixes.

https://github-production-user-asset-6210df.s3.amazonaws.com/16740832/291779848-66392282-474e-4a84-8482-d20c53c8727d.mp4

For users which will be upgrading from previous versions it is recommended to change `DEFAULT_MODEL` to `gpt-4-1106-preview` in config file `~/.config/shell_gpt/.sgptrc` since older models might not perform well with system roles. Because of significant changes in roles, unfortunately previously created custom roles and chats will not work with ShellGPT v1.0.0 and you will need to re-create them using new version.

Shoutout to all contributors: @jaycenhorton @arafatsyed @th3happybit @Navidur1 @moritz-t-w @Ismail-Ben   

<hr /><em>This discussion was created from the release <a href='https://github.com/TheR1D/shell_gpt/releases/tag/1.0.0'>1.0.0</a>.</em>

**Comment by Macmachi** [comment]
Can the API retrieve answers from the Web?

**Reply by moritztim** [reply]
- Can it use a search engine like Google to get up-to-date information?
No, you're just talking to an AI and it's talking back to you directly, without the ability to access any service. Just text back and forth.
- Does your question/prompt get sent to OpenAI's servers on the web?
Yes, that's where the AI sits. You can also host locally without sending any data to OpenAI? Yes. Read the [#LocalAI](https://github.com/TheR1D/shell_gpt#localai) section in the README for more info.

**Reply by Macmachi** [reply]
https://platform.openai.com/docs/guides/function-calling 
https://community.openai.com/t/web-search-function-using-gpt-4-turbo-model/493087

**Reply by moritztim** [reply]
That is not currently implemented afaik. You could make a feature request

**Reply by moritztim** [reply]
Altho I don't think that function calling will be implemented as the owner has often commented on feature requests that the scope of this is very simple direct text based convo.

**Reply by TheR1D** [reply]
I don't see much value in enabling web search capabilities for ShellGPT requests. With additional shell magic, similar results can be achieved. For example:
```shell
curl https://github.com/TheR1D/shell_gpt/discussions/404 | sgpt summarise
```
I have an idea for enhancing ShellGPT with function calls, although its development may take some time.
